src/query/storage/m3/dynamic_cluster.go

```diff
@@ -88,10 +96,397 @@ func NewDynamicClusterOptions() DynamicClusterOptions {
 	}
 }
 
-type dynamicCluster struct{}
+type dynamicCluster struct {
+	clusterCfgs []DynamicClusterNamespaceConfiguration
+	logger      *zap.Logger
+	iOpts       instrument.Options
+
+	sync.RWMutex
+
+	namespaces              ClusterNamespaces
```

Comment:
nit; can this have a different name and a comment maybe? `allNamepaces` or `collectedNamespaces` maybe to imply it contains the others